src/aria2p/client.py

```diff
@@ -201,14 +203,27 @@ def __init__(
         self.port = port
         self.secret = secret
         self.timeout = timeout
+        self.retries = retries
         self.listening = False
+        self.__httpx_session = None
 
     def __str__(self):
         return self.server
 
     def __repr__(self):
         return f"Client(host='{self.host}', port={self.port}, secret='********')"
 
+    def __del__(self):
+        if self.__httpx_session is not None:
+            self.__httpx_session.close()
+
+    @property
+    def _session(self):
+        if self.__httpx_session is not None:
+            return self.__httpx_session
+        self.__httpx_session = httpx.Client(timeout=self.timeout, transport=httpx.HTTPTransport(verify=False, retries=self.retries))
+        return self.__httpx_session
```

Comment:
Yeah, but i think in different way because i have tg bot.

Sometimes i don't use aria2 so I prefer creating the session after first request.

Context manager for light use is ok also

Reply:
I think the `httpx.Client` object doesn't initialize anything anyway at instantiation, or if it does, it does it when sending the first request. Its context manager is only there to clear things up when exiting. All the `__enter__` methods in `httpx` and `httpcore` only return `self` (except `BaseClient` which also changes the state from `UNOPENED` to `OPEN`).

So I think it's safe to always instantiate our HTTPX client. This won't consume any resource (apart from a tiny amount of memory I suppose).